### PR TITLE
gh-111933: fix broken link to A.Neumaier article

### PR DESCRIPTION
**Pull request by skirpichev**
It seems there are links to the archive.org in comments across the codebase.  So, I hope it's fine.

Alternatively, we could point to the journal page: https://doi.org/10.1002/zamm.19740540106 (not open access, however).  Or just add a citation (Neumaier, A. (1974), Rundungsfehleranalyse einiger Verfahren zur Summation endlicher Summen. Z. angew. Math. Mech., 54: 39-51.).

* Issue: gh-111933

### CODE REVIEW

**Comment by skirpichev** [comment]
CC @rhettinger

**Comment by JelleZijlstra** [comment]
I would prefer adding the DOI and the full citation to the comment. It's better to have more ways to find the article for those interested.

File: Python/bltinmodule.c [hunk 1]
```diff
@@ -2611,7 +2611,7 @@ builtin_sum_impl(PyObject *module, PyObject *iterable, PyObject *start)
             }
             if (PyFloat_CheckExact(item)) {
                 // Improved Kahan–Babuška algorithm by Arnold Neumaier
-                // https://www.mat.univie.ac.at/~neum/scan/01.pdf
+                // https://web.archive.org/web/20220804051351/https://www.mat.univie.ac.at/~neum/scan/01.pdf
```
Comment (skirpichev) [inline Python/bltinmodule.c]:
```suggestion
                // Neumaier, A. (1974), Rundungsfehleranalyse einiger Verfahren zur Summation endlicher Summen.
                // Z. angew. Math. Mech., 54: 39-51. https://doi.org/10.1002/zamm.19740540106
```
@JelleZijlstra ?

**Review by rhettinger — APPROVED** [review]

File: Python/bltinmodule.c (hunk 1 above)
Comment (JelleZijlstra) [reply Python/bltinmodule.c]: Let's keep them both:
```suggestion
                // Neumaier, A. (1974), Rundungsfehleranalyse einiger Verfahren zur Summation endlicher Summen.
                // Z. angew. Math. Mech., 54: 39-51. https://doi.org/10.1002/zamm.19740540106
                // https://web.archive.org/web/20220804051351/https://www.mat.univie.ac.at/~neum/scan/01.pdf
```

**Comment by skirpichev** [comment]
I've commited @JelleZijlstra suggestion above, but archive.org link was replaces with url to the prof.Neumaier homepage (https://arnold-neumaier.at/scan/01.pdf), thanks to @pochmann.

**Comment by terryjreedy** [comment]
How about adding 'https://en.wikipedia.org/wiki/2Sum'?  For me, it is a more useful link than the noisy image of German text, which as an image cannot be put into a translator.

**Comment by skirpichev** [comment]
> How about adding 'https://en.wikipedia.org/wiki/2Sum'?

Then, maybe
https://en.wikipedia.org/wiki/Kahan_summation_algorithm#Further_enhancements
?  It has relevant pseudocode and also a link (broken) to scanned text.

File: Python/bltinmodule.c [hunk 2]
```diff
@@ -2611,7 +2611,10 @@ builtin_sum_impl(PyObject *module, PyObject *iterable, PyObject *start)
             }
             if (PyFloat_CheckExact(item)) {
                 // Improved Kahan–Babuška algorithm by Arnold Neumaier
-                // https://www.mat.univie.ac.at/~neum/scan/01.pdf
+                // Neumaier, A. (1974), Rundungsfehleranalyse einiger Verfahren
+                // zur Summation endlicher Summen.  Z. angew. Math. Mech.,
+                // 54: 39-51. https://doi.org/10.1002/zamm.19740540106
+                // Scanned text: https://arnold-neumaier.at/scan/01.pdf
```
Comment (skirpichev) [inline Python/bltinmodule.c]:
```suggestion
                // https://en.wikipedia.org/wiki/Kahan_summation_algorithm#Further_enhancements
```
?

**Pull request merged**